src/key_io.cpp

```diff
@@ -270,36 +270,58 @@ std::string EncodePaymentAddress(const libzcash::PaymentAddress& zaddr)
     return boost::apply_visitor(PaymentAddressEncoder(Params()), zaddr);
 }
 
-libzcash::PaymentAddress DecodePaymentAddress(const std::string& str)
+template<typename T1, typename T2, typename T3 = T2>
+T1 DecodeAny(
+    CChainParams::Base58Type type,
+    size_t size,
```

Comment:
Use a `std::pair<> sprout` for these (so the calls to `DecodeAny` are more readable).